packages/calypso-data-stores/src/domain-suggestions/actions.ts

```diff
@@ -1,7 +1,11 @@
 /**
  * Internal dependencies
  */
-import { ActionType, DomainSuggestion, DomainSuggestionQuery } from './types';
+import {
+	DomainSuggestionsActionType as ActionType,
+	DomainSuggestion,
+	DomainSuggestionQuery,
+} from './types';
 
 export const receiveDomainSuggestions = (
```

Comment:
Is there a difference between `suggestions: Type | undefined` and marking the parameter as optional with `suggestions?: Type`?

Also, the `type: ActionType as const` forbids reassigning the `type` field?

Reply:
> the `type: ActionType as const` forbids reassigning the `type` field?

Are you referring to cases like this?

https://github.com/Automattic/wp-calypso/blob/a91b3fb89a1803e34050742c5072e83a85555084/packages/calypso-data-stores/src/verticals/actions.ts#L7

These [`const` assertions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) combine with action creators and the `enum ActionType` so that you know exactly which type of action is created. Without the `const` assertion, the inference only knows that it is of type `ActionType`, not `ActionType[ ENUM_MEMBER ]`.

For example:

```ts
const inc = () => ({ type: 'INC' as const })
const reducerInc = (s: number, a: ReturnType<typeof inc>) {
	// We know it's `INC`:
	a; // a: { type: 'INC'; }
}

const dec = () => ({ type: 'DEC' })
const reducerDec = (s: number, a: ReturnType<typeof dec>) {
	// Here we can't infer _which_ action:
	a; // a: { type: string; }
}
```

Reply:
> Are you referring to cases like this?

Yes, but I couldn't tag the line directly because GitHub won't let me if it's not part of the diff 😦 

Thanks for the explanation, it confirms the hunch I had -- that it tells the type system and code completion that the `type` field has a specific exact value, not that it's just any string.